katsdpcontroller/generator.py

```diff
@@ -1778,17 +1783,19 @@ def _make_continuum_imager(g, config, capture_block_id, name, telstate, target_m
             '--telstate', '{endpoints[telstate_telstate]}',
             '--access-key', '{resolver.s3_config[continuum][read][access_key]}',
             '--secret-key', '{resolver.s3_config[continuum][read][secret_key]}',
-            '--select', f'scans="track"; corrprods="cross"; targets=[{target.description!r}]',
-            '--capture-block-id', capture_block_id,
-            '--output-id', name,
-            '--telstate-id', telstate.join(name, target_name),
-            '--outputdir', DATA_VOL.container_path,
-            '--mfimage', _render_continuum_parameters(mfimage_parameters),
-            '-w', '/mnt/mesos/sandbox', data_url
+            '--select',
+            escape_format(f'scans="track"; corrprods="cross"; targets=[{target.description!r}]'),
```

Comment:
Might be slightly more readable to construct a distinct list for the keys that want escaping, and then iterate through that to escape them.

Reply:
Yeah, I was thinking that. I'll definitely do it for the continuum imager. For the spectral imager the arguments of the two types are currently somewhat interleaved. I feel that sorting them by need for escaping will probably make the final command-line less readable (e.g. the `-i` arguments will no longer be grouped together), and there are fewer arguments that need escaping, so I'm tempted to leave things as is.

I also realised there were some bugs because I'm escaping URLs but they actually have genuine format strings in them (for the location of telstate) and instead `_stream_url` needs to do escaping. Will get back to it tomorrow.